base/statistics.jl

```diff
@@ -630,19 +630,10 @@ function quantile!(q::AbstractArray, v::AbstractVector, p::AbstractArray;
     if size(p) != size(q)
         throw(DimensionMismatch("size of p, $(size(p)), must equal size of q, $(size(q))"))
     end
+    isempty(q) && return q
 
-    isempty(v) && throw(ArgumentError("empty data vector"))
```

Comment:
It's still there. I just split out the previously duplicated code to the `_quantilesort` function.